manage_breast_screening/notifications/management/commands/helpers/message_batch_helpers.py

```diff
@@ -59,10 +62,38 @@ def mark_batch_as_failed(
                     }
                 )
             )
+        elif response.status_code == VALIDATION_ERROR_STATUS_CODE:
+            MessageBatchHelpers.process_validation_errors(message_batch, retry_count)
         else:
             message_batch.status = MessageBatchStatusChoices.FAILED_UNRECOVERABLE.value
         message_batch.save()
 
         for message in message_batch.messages.all():
             message.status = MessageStatusChoices.FAILED.value
             message.save()
+
+    @staticmethod
+    def process_validation_errors(message_batch: MessageBatch, retry_count: int = 0):
+        message_batch_errors = json.loads(message_batch.nhs_notify_errors).get("errors")
+
+        for error in message_batch_errors:
+            message_index_result = re.search(
+                MESSAGE_PATH_REGEX, error["source"]["pointer"]
+            )
+            if message_index_result is not None:
+                message_index = int(message_index_result.group(0))
+                message = message_batch.messages.all()[message_index]
+                message.batch = None
```

Comment:
should we be creating a new Batch to add this Message to, and giving the Batch a status of FAILED_UNRECOVERABLE in order to tie it in with the same monitoring/triaging that might happen about other unrecoverable failures?

Reply:
The more I think about it we should do this, and then the MessageBatch it belongs to can also store the nhs_notify_errors

Reply:
done! also added another test for having multiple validation errors which uncovered some flaws in how the messages were being iterated